cf_units/__init__.py

```diff
@@ -781,7 +781,7 @@ def as_unit(unit):
         if use_cache:
             result = _CACHE.get(unit)
         if result is None:
-            result = Unit(unit)
+            result = Unit(unit, calendar=getattr(unit, 'calendar', None))
```

Comment:
Not that I'm aware of, [see](https://github.com/SciTools/cf_units/blob/24b8774fa4e4d1502dfed63812c04db64dd84be9/cf_units/__init__.py#L907).

To create a Unit object, the user passes a 'unit' to the class (which is turned into a string) alongside a calendar (and that's all).